xbmc/input/KeyboardLayout.cpp

```diff
@@ -69,7 +69,7 @@ bool CKeyboardLayout::Load(const TiXmlElement* element)
 
   const TiXmlElement *keyboard = element->FirstChildElement("keyboard");
   if (element->Attribute("codingtable"))
-    m_codingtable = CInputCodingTableFactory::CreateCodingTable(element->Attribute("codingtable"), element);
+    m_codingtable = IInputCodingTablePtr(CInputCodingTableFactory::CreateCodingTable(element->Attribute("codingtable"), element));
```

Comment:
`std::make_shared` won't work here because this isn't a call to a constructor but a factory.